pkg/kubelet/kuberuntime/security_context.go

```diff
@@ -95,6 +95,42 @@ func verifyRunAsNonRoot(pod *v1.Pod, container *v1.Container, uid *int64, userna
 	}
 }
 
+func verifyRunAsNonRootGroup(pod *v1.Pod, container *v1.Container, gid *int64, groupname string) error {
```

Comment:
Echo what @tallclair mentioned in the previous comment. I *think * this validation comes later and volumes may have already be mounted with FSGroup earlier. Please check if that's true, and having the validation check prior to the volume mount AND at the apiserver is preferred.

Reply:
@yujuhong thanks will verify.

Reply:
@yujuhong even if the volumes are already mounted, failing this check will not start the pod right ? I will add an e2e validation, but am i missing something here ?

Reply:
Yes, containers will not start. I prefer having some check so that kubelet could avoid mounting in this case if possible. Validation at the apiserver is one way to achieve that.

Reply:
@yujuhong @tallclair validation at the apiserver will not catch the case where gid is hardcoded in the docker image itself. This validation catches that scenario. Would you prefer that we add this validation at both apiserver level and also do the image part of the validation here ? This way the apiserver validation fails the pod creation (and hence mounting) and if the docker image has the gid hardcoded as 0, then that validation happens at runtime.